### DISCUSSION TITLE
how to perform reciprocal space mapping

### DISCUSSION THREAD

**Discussion by dragonyanglong** [dkriegner/xrayutilities#199] [General]
Hi all,

I have a set of single crystal images collected while rotating 360 degrees. I would like to transform them into reciprocal space volume. So it includes finding the orientation matrix, and then perform the transformation of all 360 images. Is there any appropriate example code for doing this? Thanks.

**Comment by dkriegner** [comment]
Unfortunately the answer is no. The current examples cover only parts of what you want to achieve. In particular there are some examples which cover how to set up the goniometer and area detector parts. see for example https://github.com/dkriegner/xrayutilities/blob/main/examples/xrayutilities_example_plot_3D_ESRF_ID01.py

For finding the orientation matrix for your case there are no examples.
If you generate a nice example it would be worth to add it to the collection of examples.

**Comment by dragonyanglong** [comment]
@dkriegner , thanks for the reply. I can try to set up an example like this. Could you point me to the functions related with finding the orientation matrix, and transform into reciprocal space?

**Comment by dkriegner** [comment]
There is not much regarding the orientation matrix. The biggest problem I see here is that you will have to identify and index your peaks.  There is nothing in xrayutilities which particularly is suited for that.
Once you have your peaks identified its rather straight forward. see for example https://github.com/dkriegner/xrayutilities/blob/main/examples/xrayutilities_orientation_matrix.py

the transformation to reciprocal space you find in the example i mentioned in my previous report. you will have to define your goniometer and detector parameters. this part should be the easy one.
